### PR TITLE
feat: support ResponseHeaderModifier filter

### PR DESCRIPTION
**Pull request by rodman10**
**What this PR does / why we need it**:

This PR support ResponseHeaderModifier Gateway API filter.

**Which issue this PR fixes**:
Fixes https://github.com/Kong/kubernetes-ingress-controller/issues/3683

**Special notes for your reviewer**:

**PR Readiness Checklist**:

Complete these before marking the PR as `ready to review`:

- [x] the `CHANGELOG.md` release notes have been updated to reflect any significant (and particularly user-facing) changes introduced by this PR

### CODE REVIEW

**Comment by rodman10** [comment]
Through translating `responseHeaderModifier` field to `response-transform` Kong plugin with `headers` config, the extended [ResponseHeaderModifier](https://gateway-api.sigs.k8s.io/geps/gep-1323/) gateway-api filter is now supported.
Since `HTTPRouteFilter` in `HTTPBackendRef` is Implementation-specific support, same as other implemented filter, `ResponseHeaderModifier` will only support in `HTTPRouteRule`.
This PR contains following changes:
- add new helper function for both request and response header modifier filter.
- complement request header modifier unittest.

File: internal/dataplane/parser/translators/httproute_test.go [hunk 1]
```diff
@@ -33,6 +33,12 @@ func TestGeneratePluginsFromHTTPRouteFilters(t *testing.T) {
 								Value: "bar",
 							},
 						},
+						Add:  []gatewayv1beta1.HTTPHeader{
```
Comment (randmonkey) [inline internal/dataplane/parser/translators/httproute_test.go:36]: As linter suggested, only one space should be reserved here.

Comment (rodman10) [reply internal/dataplane/parser/translators/httproute_test.go]: Thanks, I has just fix it.

**Review by randmonkey — APPROVED** [review]

**Review by mlavacca — APPROVED** [review]
Looks great @rodman10! Thanks for your contribution 🚀

**Pull request merged**